pdctl/command/config_command.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	configPrefix   = "pd/api/v1/config"
+	schedulePrefix = "pd/api/v1/config/schedule"
+)
+
+// NewConfigCommand return a config subcommand of rootCmd
+func NewConfigCommand() *cobra.Command {
+	conf := &cobra.Command{
+		Use:   "config <subcommand>",
+		Short: "tune pd configs",
+	}
+	conf.AddCommand(NewShowConfigCommand())
+	conf.AddCommand(NewSetConfigCommand())
+	return conf
+}
+
+// NewShowConfigCommand return a show subcommand of configCmd
+func NewShowConfigCommand() *cobra.Command {
+	sc := &cobra.Command{
+		Use:   "show",
+		Short: "show config fo PD",
+		Run:   configShowCommandF,
+	}
+	return sc
+}
+
+// NewSetConfigCommand return a set subcommand of configCmd
+func NewSetConfigCommand() *cobra.Command {
+	sc := &cobra.Command{
+		Use:   "set <option> <value>",
+		Short: "set the option with value",
+		Run:   configSetCommandF,
+	}
+	return sc
+}
+
+func configShowCommandF(cmd *cobra.Command, args []string) {
+	url := getAddressFromCmd(cmd, schedulePrefix)
+	r, err := http.Get(url)
+	if err != nil {
+		fmt.Printf("Failed to get config:[%s]\n", err)
+		return
+	}
+	defer r.Body.Close()
+	if r.StatusCode != http.StatusOK {
+		fmt.Println(r.StatusCode, "error")
+		return
+	}
+
+	io.Copy(os.Stdout, r.Body)
```

Comment:
r.Body is an io.Reader . here use io.Copy is ok. I have checked the response statusCode.